### PR TITLE
Fetch Staff Profile By Id

### PR DESCRIPTION
**Pull request by sahitya-desireddy**
Fetch Staff Profile By Id

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by vasthav-patta — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java [hunk 1]
```diff
@@ -414,5 +416,44 @@ public ResponseEntity<StaffProfileCreationResponse> updateStaffUserProfile(@Requ
...
     }
 
+    @ApiOperation(
+            value = "This API search a staff user by Id",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+    )
+    @ApiResponses({
```
Comment (manukundloo-hmcts) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java:426]: Hi @sahitya-desireddy just make The Swagger Documentation as per the code . Here the response code should be match with the code (200).

Comment (sahitya-desireddy) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java:425]: Thanks @manukundloo-hmcts resolved now please review

**Review by manukundloo-hmcts — DISMISSED** [review]

**Review by vasthav-patta — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        String path = "/profile/search-by-id";
+        String searchString = "?id=" + caseWorkerId;
+        caseworkerReferenceDataClient.setBearerToken(null);
+
+        userProfileGetUserByIdWireMock(caseWorkerId, 200);
+        SearchStaffUserByIdResponse getResponse =  (SearchStaffUserByIdResponse)caseworkerReferenceDataClient
+                .fetchStaffUserById(SearchStaffUserByIdResponse.class, path + searchString, ROLE_STAFF_ADMIN);
+        Assertions.assertNotNull(getResponse);
+        Assertions.assertEquals(caseWorkerId, getResponse.getCaseWorkerId());
+        Assertions.assertNotNull(getResponse.getIdamStatus());
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java:130]: assert expected idam status

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java (hunk 2 above)
Comment (sahitya-desireddy) [reply src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java:130]: added Thank you @rajkatla-hmcts

**Review by manukundloo-hmcts — DISMISSED** [review]

**Review by rajkatla-hmcts — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        Assertions.assertEquals("pending", getResponse.getIdamStatus());
+
+    }
+
+}
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java:135]: Functional test is missing for this new endpoint, as discussed pls add one

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java (hunk 3 above)
Comment (sahitya-desireddy) [reply src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java:135]: As discussed with @kavithagade12 this functional test we are covering in the new ticket ... hence resolving this review comment as now will create new ticket  for this test ..fyi @kavithagade12

**Review by prudhvi-maddineni — APPROVED** [review]

**Review by vasthav-patta — DISMISSED** [review]

**Review by rajkatla-hmcts — APPROVED** [review]

**Review by vasthav-patta — APPROVED** [review]

**Pull request merged**